src/postgres/commands/setPostgresCredentials.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ext } from "../../extensionVariables";
+
+interface IPersistedServer {
+    id: string;
+    username: string;
+}
+
+export async function setPostgresCredentials(username: string, password: string, serverId: string): Promise<void> {
+    if (ext.keytar) {
+        const serviceName: string = "ms-azuretools.vscode-cosmosdb.postgresPasswords";
```

Comment:
I see you copied this code from `PostgresServerTreeItem`, but I think you forgot to delete the original code from that file

Reply:
I haven't completed all the changes yet. I was hoping to get feedback on these new execute steps that is if I am going in the right direction from what we discussed, before I make rest of the changes. I will push rest of the changes then

Reply:
Yep the execute steps look good overall

Reply:
I was planning to leave the setFirewallRule function as is in configurePostgresFirewall.ts even though duplicate code exists in the execute wizard ...SetFirewallStep. What do you think, should I remove that and make changes to configurePostgresFirewall.ts as well?

Reply:
Yes please remove the duplicate code. I believe minimal changes should be required to re-use this function

Reply:
Do you mean reverting only configurePostgresFirewall.ts or reverting configurePostgresFirewall.ts and removing PostgresServerSetFirewallStep.ts as well?

Reply:
I'm only talking about changes you made in [this last commit](https://github.com/microsoft/vscode-cosmosdb/pull/1404/commits/d27d1606cf37be50ef5d445a933e2c3c22e7acb9)

So you should definitely keep `PostgresServerSetFirewallStep.ts`, but as an example you should change this:
```
const resourceGroup: string = nonNullProp(nonNullProp(wizardContext, 'resourceGroup'), 'name');
```

back to this:

```
const resourceGroup: string = nonNullProp(wizardContext, 'newResourceGroupName');
```

since that change was related to [this comment](https://github.com/microsoft/vscode-cosmosdb/pull/1404#discussion_r414127094)